crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+use move_core_types::language_storage::StructTag;
+use move_core_types::value::{self, MoveStruct, MoveTypeLayout};
+use sui_indexer::models_v2::objects::StoredObject;
+use sui_types::dynamic_field::DynamicFieldInfo;
+use sui_types::{dynamic_field::DynamicFieldType, TypeTag};
+
+use super::{
+    base64::Base64, move_object::MoveObject, move_value::MoveValue, sui_address::SuiAddress,
+};
+use crate::context_data::package_cache::PackageCache;
+use crate::error::{code, Error};
+use crate::{context_data::db_data_provider::PgManager, error::graphql_error};
+use sui_types::object::Object as NativeSuiObject;
+
+pub(crate) struct DynamicField {
+    pub stored_object: StoredObject,
+}
+
+impl DynamicField {
+    pub fn is_dynamic_object(&self) -> bool {
```

Comment:
nit: do these need to be `pub`? I'm hoping they can at least be `pub(crate)` but should possible to make private.

Also nit: Can you put the public interface at the top (in this case the `impl` block that's annotated with `#[Object]`)?

Reply:
I dont need these anymore thanks to the suggestion above to keep some unique fields on DynamicField